src/mono/mono/utils/mono-threads-wasm.c

```diff
@@ -37,15 +37,17 @@ wasm_get_stack_size (void)
 static int
 wasm_get_stack_base (void)
 {
-	g_assert_not_reached ();
-	return 0;
+	// TODO: For WASI, we need to ensure the stack location makes sense and won't interfere with the heap.
+	// Currently these hardcoded values are sufficient for a working prototype.
+	return 1;
```

Comment:
I think you should make this an aligned offset instead of 1, there is code that breaks if stack frames and mallocs don't come back at aligned offsets (an alignment of 4 is probably enough)

Reply:
OK, done.